libsolidity/interface/CompilerStack.cpp

```diff
@@ -262,7 +262,9 @@ bool CompilerStack::compile()
 	for (Source const* source: m_sourceOrder)
 		for (ASTPointer<ASTNode> const& node: source->ast->nodes())
 			if (auto contract = dynamic_cast<ContractDefinition const*>(node.get()))
-				compileContract(*contract, compiledContracts);
+				/// Only compile if contract name is requested or no set is supplied.
+				if (m_requestedContractNames.empty() || m_requestedContractNames.count(contract->name()))
```

Comment:
Can you change this to `m_requestedContractNames.count(contract->fullyQualifiedName()) || m_requestedContractNames.count(contract->name())`?

Reply:
Hm, this might actually be a little trickier in case of inheritance and contracts creating other contracts.

Reply:
I don't see a problem there. If `C` creates an instance of `D`, just compiling `C` will trigger `D` as a dependency.

Reply:
Why will it trigger D as a dependency? Can you check what the compiler does if you compile code
`contract D{} contract C { function f() { new D(); } }` and only requesting compilation of C?

Reply:
Added specific tests for it, all pass.